datalab/mlalpha/_dataset.py

```diff
@@ -59,21 +60,34 @@ def __init__(self, files, schema=None, schema_file=None):
     else:
       with ml.util._file.open_local_or_gcs(schema_file, 'r') as f:
         self._schema = json.load(f)
+        
     if isinstance(files, basestring):
       files = [files]
-    self._files = []
-    for file in files:
-      # glob_files() returns unicode strings which doesn't make DataFlow happy. So str().
-      self._files += [str(x) for x in ml.util._file.glob_files(file)]
+    self._input_files = files
+    
+    self._glob_files = []
+
+
+  @property
+  def _input_files(self):
+    """Returns the file list that was given to this class without globing files."""
+    return self._input_files
 
   @property
   def files(self):
-    return self._files
+    if self._glob_files:
```

Comment:
Is:

if not self._glob_files:
  self._glob_files += ...

return self._glob_files

better?